lib/src/presentation/widget/inherited_bloc.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:flutter/widgets.dart';
+
+import '../../core/bloc/i_bloc.dart';
+
+class InheritedBloc<T extends IBloc> extends InheritedWidget {
```

Comment:
Why do you need this class?

Reply:
Not so much need, but prefer. We could just pass the bloc as a parameter down the tree, but this is a bit unwieldy. Using an Inherited widget makes accessing this data more comfortable, and combining it with the BlocProvider gives us a good place to call the bloc's dispose method

Reply:
Why do you think that is a bit unwieldy?

Reply:
The movie list widgets needed to call the fetch more movies method, so they needed to have the bloc as a parameter. But passing the bloc solely to call a method felt awkward. This way we can pass to the movie list widgets only the list of movies to display, which made more sense to me. It also means we dont need to pass the bloc to the views, we can just place an inherited bloc at the root of the tree and all of the widgets can access it. We were going to need a stateful widget at that root to dispose of the bloc, so we might as well also propagate that bloc downwards.

Reply:
But the Bloc is the connection between Presentation and domain layer, so the view need to have a bloc, and here you only have one method, but you can call more than one

Reply:
hmm i see. I get the part about the bloc, it does make sense for the view to have it. I'm thinking of moving the bloc to the view but not as a parameter, rather have the view instantiate the bloc it needs. This will make the view creation a lot shorter since it just needs the repository as a parameter, and it can safely call bloc.dispose(), which makes sense. In fact, since each type of bloc is coupled to a particular type of usecase, i think it might be a better idea to instantiate the usecase inside the bloc instead of getting it as a parameter. Since it doesn't really make sense for the use case to exist outside of the bloc, and the only thing we mock is the repo which is still going to be a parameter.

I didn't understand what you meant by "can call more than one method" though.

Reply:
ended up going in this direction. Now the views only take a usecase and handle creation and disposal of the bloc